main/lsp/LSPMessage.cc

```diff
@@ -167,6 +166,9 @@ bool LSPMessage::isDelayable() const {
         case LSPMethod::TextDocumentPublishDiagnostics:
         case LSPMethod::SorbetShowOperation:
         case LSPMethod::SorbetTypecheckRunInfo:
+        // This request involes a specific file location and SHOULD never be delayed but we need completion to be fast.
+        // TODO(jez) Check if we still need this after pre-emptible slow path
+        case LSPMethod::TextDocumentCompletion:
```

Comment:
Can you audit `completion.cc` to ensure that you guard against invalid file locations? Now that these requests can desync, it's possible to get a completion location that no longer exist. Acting on it will cause crashes.